test/integration_test/basic_test.go

```diff
@@ -308,31 +321,128 @@ func BasicCsiRegression(tc *types.TestCase) func(*testing.T) {
+// 6. Delete "stork-scheduler" pods and validate they get re-deployed
+// 7. Enable Stork webhook-controller and validate
+// 8. Disable Stork webhook-controller and validate
+// 9. Remove Stork webhook-controller and validate
+// 10. Enable hotNetwork and validate
+// 11. Disable hostNetwork and validate
+// 12. Remove hostNetwork and valiate
+// 13. Disable Stork and validate Stork components got successfully removed
+// 14. Enabled Stork and validate Stork components and images
+// 15. Delete StorageCluster and validate it got successfully removed
+func BasicStorkRegression(tc *types.TestCase) func(*testing.T) {
+	return func(t *testing.T) {
+		if tc.ShouldSkip() {
+			t.Skip()
+		}
 
-		// Sleep for 20 seconds to let operator start the update process
-		logrus.Debug("Sleeping for 20 seconds...")
-		time.Sleep(20 * time.Second)
+		var err error
+		testSpec := tc.TestSpec(t)
+		cluster, ok := testSpec.(*corev1.StorageCluster)
+		require.True(t, ok)
 
-		logrus.Infof("Validate StorageCluster %s", cluster.Name)
-		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateUpdateTimeout, ci_utils.DefaultValidateUpdateRetryInterval, true, "")
+		// Create and validate StorageCluster
+		cluster = ci_utils.DeployAndValidateStorageCluster(cluster, ci_utils.PxSpecImages, t)
+
+		// Validate Stork is enabled by default
+		if !cluster.Spec.Stork.Enabled {
+			require.NoError(t, fmt.Errorf("failed to validate Stork is enabled by default, it should be enabled, but it is set to %v", cluster.Spec.Stork.Enabled))
+		}
+
+		// Validate Webhook controller arg doesn't by default
+		if len(cluster.Spec.Stork.Args["webhook-controller"]) != 0 {
+			require.NoError(t, fmt.Errorf("failed to validate webhook-controller, it shouldn't exist by default, but it is set to %s", cluster.Spec.Stork.Args["webhook-controller"]))
+		}
+
+		// Validate HostNetwork is <nil> by default
+		if cluster.Spec.Stork.HostNetwork != nil {
+			require.NoError(t, fmt.Errorf("failed to validate HostNetwork, it shouldn't exist by default, but it is set to %v", *cluster.Spec.Stork.HostNetwork))
+		}
+
+		logrus.Info("Delete stork pods and validate they get re-deployed")
+		err = appsops.Instance().DeleteDeploymentPods("stork", cluster.Namespace, 120*time.Second)
+		require.NoError(t, err)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
 		require.NoError(t, err)
 
-		logrus.Info("Enable CSI and validate StorageCluster")
+		logrus.Info("Delete stork-scheduler pods and validate they get re-deployed")
+		err = appsops.Instance().DeleteDeploymentPods("stork-scheduler", cluster.Namespace, 120*time.Second)
+		require.NoError(t, err)
+		err = testutil.ValidateStorageCluster(ci_utils.PxSpecImages, cluster, ci_utils.DefaultValidateDeployTimeout, ci_utils.DefaultValidateDeployRetryInterval, true, "")
+		require.NoError(t, err)
```

Comment:
Let's club these two together. Delete both stork and scheduler pods and then validate.

Reply:
Done